rotkehlchen/chain/evm/decoding/curve/curve_cache.py

```diff
@@ -388,8 +408,21 @@ def _query_curve_data_from_chain(
         deployed_block=0,  # deployment_block is not used and the contract is dynamic
     )
     pool_count = metaregistry.call(node_inquirer=evm_inquirer, method_name='pool_count')
-    new_pools = []
+    new_pools: list[CurvePoolData] = []
+    last_notified_ts = 0
     for pool_index in range(pool_count):
+        if (current_time := ts_now()) - last_notified_ts > 5:
+            msg_aggregator.add_message(
+                message_type=WSMessageType.PROTOCOL_CACHE_UPDATES,
+                data={
+                    'protocol': CPT_CURVE,
+                    'chain': evm_inquirer.chain_id.to_name(),
+                    'processed': 0,
+                    'total': len(new_pools),
+                },
+            )
+            last_notified_ts = current_time
```

Comment:
this logic can be abstracted. Create a function `notify_cache_query_status` that takes as argument the counterparty, the current processed, the total and the chain name and then you do

```
last_notified_ts = notify_cache_query_status(...)
```